pkg/controller/keyvault/ingress_secret_provider_class.go

```diff
@@ -170,5 +175,10 @@ func (i *IngressSecretProviderClassReconciler) buildSPC(ing *netv1.Ingress, spc
 			"objects":                string(objects),
 		},
 	}
+
+	if i.config.Cloud != "" {
```

Comment:
can we add validation in the config package to ensure that the cloud name provided is a valid one?

Reply:
We can't import available cloud options from somewhere so that means we'd have to hardcode them and update that array every time a new cloud is added. That's a lot of work and we really shouldn't hardcode something like that which would result in WAR completely breaking in a region unless we add it manually (lots of manual work).

We instead validate that RP is passing proper information in through `Cloud` with unit and e2e tests. Operator validation doesn't help us

Reply:
that makes sense, can + should have it in RP